languages/C.groovy

```diff
@@ -0,0 +1,379 @@
+    }
+
+
+    // Get logical file
+    LogicalFile logicalFile = resolverUtils.createLogicalFile(dependencyResolver, buildFile)
+
+    // create mvs commands
+    String member = CopyToPDS.createMemberName(buildFile)
+    File logFile = new File( props.userBuild ? "${props.buildOutDir}/${member}.log" : "${props.buildOutDir}/${member}.c.log")
+    if (logFile.exists())
+        logFile.delete()
+    MVSExec compile = createCompileCommand(buildFile, logicalFile, member, logFile)
+    MVSExec linkEdit = createLinkEditCommand(buildFile, logicalFile, member, logFile)
+
+    // execute mvs commands in a mvs job
+    MVSJob job = new MVSJob()
+    job.start()
+
+    // compile the c program
+    int rc = compile.execute()
+    int maxRC = props.getFileProperty('cc_compileMaxRC', buildFile).toInteger()
+
+    boolean bindFlag = true
+
+    if (rc > maxRC) {
+        bindFlag = false
+        String errorMsg = "*! The compile return code ($rc) for $buildFile exceeded the maximum return code allowed ($maxRC)"
+        println(errorMsg)
+        props.error = "true"
+        buildUtils.updateBuildResult(errorMsg:errorMsg,logs:["${member}.log":logFile])
+    }
+    else { // if this program needs to be link edited . . .
+
+        // Store db2 bind information as a generic property record in the BuildReport
+
+        String needsLinking = props.getFileProperty('cc_linkEdit', buildFile)
+        if (needsLinking.toBoolean()) {
+            rc = linkEdit.execute()
+            maxRC = props.getFileProperty('cc_linkEditMaxRC', buildFile).toInteger()
+
+            if (rc > maxRC) {
+                bindFlag = false
+                String errorMsg = "*! The link edit return code ($rc) for $buildFile exceeded the maximum return code allowed ($maxRC)"
+                println(errorMsg)
+                props.error = "true"
+                buildUtils.updateBuildResult(errorMsg:errorMsg,logs:["${member}.log":logFile])
+            }
+            else {
+                if(!props.userBuild){
+                    // only scan the load module if load module scanning turned on for file
+                    String scanLoadModule = props.getFileProperty('cc_scanLoadModule', buildFile)
+                    if (scanLoadModule && scanLoadModule.toBoolean())
+                        impactUtils.saveStaticLinkDependencies(buildFile, props.linkedit_loadPDS, logicalFile)
+                }
+            }
+        }
+    }
+
+    // clean up passed DD statements
+    job.stop()
+}
+
+// end script
+
+
+//********************************************************************
+//* Method definitions
+//********************************************************************
+
+/*
+ * createCParms - Builds up the c compiler parameter list from build and file properties
+ */
+def createCParms(String buildFile, LogicalFile logicalFile) {
+    def parms = props.getFileProperty('cc_compileParms', buildFile) ?: ""
```

Comment:
@saipriya-m Your PR is missing a `C.properties` file that defines the Compile and link options the language script that is typically included in the `application-conf` directory.

Reply:
@dennis-behm is there any different repo where i can push the application-conf/C.properties file?

Reply:
Hi, within the zAppBuild `samples/application-conf` dir at https://github.com/IBM/dbb-zappbuild/tree/a719842f8b9738133e375eaade8497289c30fc05/samples/application-conf, we maintain all application-cons samples, including the README of the parameters.